linode/helper/instance.go

```diff
@@ -199,34 +199,58 @@ func ExpandInterfaceIPv4(ipv4 any) *linodego.VPCIPv4 {
 	if vpcAddress == "" && nat1To1 == "" {
 		return nil
 	}
-	return &linodego.VPCIPv4{
-		VPC:     vpcAddress,
-		NAT1To1: nat1To1,
+
+	result := &linodego.VPCIPv4{
+		VPC: vpcAddress,
+	}
+
+	if nat1To1 != "" {
+		result.NAT1To1 = &nat1To1
 	}
+
+	return result
 }
 
 func ExpandConfigInterface(ifaceMap map[string]interface{}) linodego.InstanceConfigInterfaceCreateOptions {
+	purpose := linodego.ConfigInterfacePurpose(ifaceMap["purpose"].(string))
 	result := linodego.InstanceConfigInterfaceCreateOptions{
-		Purpose:     linodego.ConfigInterfacePurpose(ifaceMap["purpose"].(string)),
-		Label:       ifaceMap["label"].(string),
-		IPAMAddress: ifaceMap["ipam_address"].(string),
-		Primary:     ifaceMap["primary"].(bool),
+		Purpose: purpose,
+		Primary: ifaceMap["primary"].(bool),
 	}
-	if ifaceMap["subnet_id"] != nil {
-		subnet_id := ifaceMap["subnet_id"].(int)
-		if subnet_id != 0 {
-			result.SubnetID = &subnet_id
+
+	if purpose == linodego.InterfacePurposeVLAN {
+		if ifaceMap["ipam_address"] != nil {
+			result.IPAMAddress = ifaceMap["ipam_address"].(string)
 		}
-	}
 
-	if ifaceMap["ipv4"] != nil {
-		ipv4 := ifaceMap["ipv4"].([]any)
-		if len(ipv4) > 0 {
-			result.IPv4 = ExpandInterfaceIPv4(ipv4[0])
+		if ifaceMap["label"] != nil {
+			result.Label = ifaceMap["label"].(string)
 		}
 	}
-	if ifaceMap["ip_ranges"] != nil {
-		result.IPRanges = ExpandStringList(ifaceMap["ip_ranges"].([]interface{}))
+
+	if purpose == linodego.InterfacePurposeVPC {
+		if ifaceMap["subnet_id"] != nil {
+			subnet_id := ifaceMap["subnet_id"].(int)
+			if subnet_id != 0 {
+				result.SubnetID = &subnet_id
+			}
+		}
+
+		if ifaceMap["ipv4"] != nil {
+			ipv4 := ifaceMap["ipv4"].([]any)
+			if len(ipv4) > 0 {
+				result.IPv4 = ExpandInterfaceIPv4(ipv4[0])
+			}
+		}
+		if ifaceMap["ip_ranges"] != nil {
+			ranges := ifaceMap["ip_ranges"].([]interface{})
+
+			// this is for keep result.IPRanges as a nil value rather than a value of empty slice
+			// when there is not a range.
+			if len(ranges) > 0 {
+				result.IPRanges = ExpandStringList(ranges)
+			}
+		}
```

Comment:
optional: These can be condensed down a bit

```suggestion
		if ifaceMap["subnet_id"] != nil {
			if subnetId := ifaceMap["subnetId"].(int); subnetId != 0 {
				result.SubnetID = &subnetId
			}
		}

		if ifaceMap["ipv4"] != nil {
			if ipv4 := ifaceMap["ipv4"].([]any); len(ipv4) > 0 {
				result.IPv4 = ExpandInterfaceIPv4(ipv4[0])
			}
		}
		if ifaceMap["ip_ranges"] != nil {
			// this is for keep result.IPRanges as a nil value rather than a value of empty slice
			// when there is not a range.
			if ranges := ifaceMap["ip_ranges"].([]interface{}); len(ranges) > 0 {
				result.IPRanges = ExpandStringList(ranges)
			}
		}
```